btr-api/src/btr_api/services/submission.py

```diff
@@ -75,12 +76,45 @@ def create_submission(submission_dict: dict, submitter_id: int) -> SubmissionMod
 
         """
         submission = SubmissionModel.find_by_business_identifier(submission_dict['businessIdentifier'])
-        if not submission:
-            submission = SubmissionModel()
-            submission.business_identifier = submission_dict['businessIdentifier']
```

Comment:
I think this line got accidentally merged incorrectly, we need to set identifier for submission too after creating new model.